engine/src/main/java/graphql/nadel/engine/execution/NadelExecutionStrategy.java

```diff
@@ -121,7 +123,19 @@ public NadelExecutionStrategy(List<Service> services,
         List<CompletableFuture<OneServiceExecution>> result = new ArrayList<>();
         for (MergedField mergedField : fieldSubSelection.getMergedSelectionSet().getSubFieldsList()) {
             ExecutionStepInfo fieldExecutionStepInfo = executionStepInfoFactory.newExecutionStepInfoForSubField(executionCtx, mergedField, rootExecutionStepInfo);
-            Service service = getServiceForFieldDefinition(fieldExecutionStepInfo.getFieldDefinition());
+
+            boolean usesDynamicService = fieldExecutionStepInfo.getFieldDefinition().getDirectives()
+                    .stream()
+                    //TODO: Move "dynamicService" to constant somewhere
```

Comment:
I haven't found a good central place to put the directive name to ensure consistency. Perhaps `NadelDirectives` is good candidate? Although the consts there are package protected so we would have to change that.

Reply:
change it

Reply:
We could also create helper methos there

`boolean usesDynamicService = NadelDirectives.hasDirective("name",fieldExecutionStepInfo.getFieldDefinition())`